src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,33 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
+
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                Thread.sleep(1000); // Simulated delay
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+
+            UserData userData = new UserData(userId, "Details for " + userId);
+            cache.put(userId, userData); // Cache the result
```

Comment:
Access to the shared `cache` map should be synchronized to ensure thread safety. While `ConcurrentHashMap` handles individual operations atomically, the check-then-act sequence here (`containsKey` followed by `put`) is not atomic and could lead to race conditions. Consider using `computeIfAbsent` to atomically compute and store the `UserData` if absent.